polar-core/src/partial/constraints.rs

```diff
@@ -701,18 +753,90 @@ mod test {
         let mut q = p.new_query_from_term(term!(call!("lhs", [partial!("a")])), false);
         assert_partial_expression!(next_binding(&mut q)?, "a", "_this = 1");
         assert_partial_expression!(next_binding(&mut q)?, "a", "_this = 2");
-        assert!(matches!(q.next_event()?, QueryEvent::Done { .. }));
+        assert_query_done!(q);
 
         // Inverting an `in` produces a conjunction of the inverted disjunctive constraints.
         let mut q = p.new_query_from_term(term!(call!("not_lhs", [partial!("a")])), false);
         assert_partial_expression!(next_binding(&mut q)?, "a", "_this != 1 and _this != 2");
-        assert!(matches!(q.next_event()?, QueryEvent::Done { .. }));
+        assert_query_done!(q);
 
-        // Partials are not allowed on the RHS of `in`.
-        let mut q = p.new_query_from_term(term!(call!("rhs", [partial!("a")])), false);
-        let error = q.next_event().unwrap_err();
-        assert!(matches!(error, PolarError {
-            kind: ErrorKind::Runtime(RuntimeError::TypeError { .. }), ..}));
+        Ok(())
+    }
+
+    #[test]
+    fn test_contains_partial() -> TestResult {
+        let p = Polar::new();
+        p.load_str(
+            r#"
+            contains(x, y) if x in y;
+            contains_dot(x, y) if x in y.foo;
+            contains_dot_dot(x, y) if x in y.foo.bar and y.foo = 2;
+        "#,
```

Comment:
matches the formatting elsewhere

```suggestion
            r#"contains(x, y) if x in y;
               contains_dot(x, y) if x in y.foo;
               contains_dot_dot(x, y) if x in y.foo.bar and y.foo = 2;"#,
```